src/coord/src/coord.rs

```diff
@@ -687,9 +687,12 @@ impl Coordinator {
                 tx.send(Ok(ExecuteResponse::CreatedSink { existed: false }), session);
             }
             Err(e) => {
-                self.catalog_transact(vec![catalog::Op::DropItem(id)])
-                    .await
-                    .expect("deleting placeholder sink cannot fail");
+                // Drop the placeholder sink if still present.
```

Comment:
I think it'd be nice to make this look a bit like the `else` branch above and say something like "Another session may have dropped the placeholder sink while we were attempting to create the connector, in which case we don't need to do anything."